cli/cmd/encore/eject.go

```diff
@@ -0,0 +1,86 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"runtime"
+
+	"github.com/spf13/cobra"
+
+	daemonpb "encr.dev/proto/encore/daemon"
+)
+
+func init() {
+	ejectCmd := &cobra.Command{
+		Use:   "eject",
+		Short: "eject provides ways to eject your application to migrate away from using Encore",
+	}
+
+	var push bool
+	dockerEjectCmd := &cobra.Command{
+		Use:   "docker IMAGE_TAG",
+		Short: "docker builds a portable docker image of your Encore application",
+		Args:  cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			appRoot, _ := determineAppRoot()
+			imgTag := args[0]
+			dockerEject(appRoot, imgTag, push)
+		},
+	}
+
+	dockerEjectCmd.Flags().BoolVarP(&push, "push", "p", false, "push image to remote repository")
+
+	rootCmd.AddCommand(ejectCmd)
+	ejectCmd.AddCommand(dockerEjectCmd)
+}
+
+func dockerEject(appRoot string, imageTag string, push bool) {
+	interrupt := make(chan os.Signal, 1)
+	signal.Notify(interrupt, os.Interrupt)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	go func() {
+		<-interrupt
+		cancel()
+	}()
+
+	daemon := setupDaemon(ctx)
+	params := &daemonpb.DockerExportParams{}
+	if push {
+		params.PushDestinationTag = imageTag
+	} else {
+		params.LocalDaemonTag = imageTag
+	}
+
+	goos, goarch := "linux", runtime.GOARCH
```

Comment:
Should we support a buildx style docker image which has native support for a matrix of targets?

We do this for emissary and the placeholder service so they can be deployed onto either ARM64 or AMD64 systems: https://github.com/encoredev/placeholder-service/blob/main/.github/workflows/cd.yml#L64-L73

Reply:
I thought the Go compiler can target any GOOS / GOARCH from any other GOOS / GOARCH version - so the encore-go installed should be able to compile both versions without issue. (Unless I'm missing something)